### PR TITLE
refactor: get rid of DummyClientExt

### PR DESCRIPTION
**Pull request by dpc**
Now that `ClientModule` has access to its context (`Client`), it can use it instead of exposing `*Ext` trait.

I picked dummy module as a first one to refactor, because I thought it will be the easiest. Didn't realize it's used all over the tests...

Due to lifetimes issues in async context my original idea of exposing only weak pointer didn't work, but I also realized that we probably don't want to expose the whole `Client` anyway, and we should be more selective about what we allow, so decided to just delegate `Client` methods I need on the `ClientContext` for now.

While working on it I noticed lots of possible improvements, and don't want to waste time splitting things into smaller commits, etc. so just did what I need immediate to make things simpler. Will follow through with some more changes.

### CODE REVIEW

File: fedimint-client/src/lib.rs [hunk 1]
```diff
@@ -564,13 +564,26 @@ const SUPPORTED_CORE_API_VERSIONS: &[fedimint_core::module::ApiVersion] =
...
+impl<'m, M> ops::Deref for ClientModuleInstance<'m, M>
+where
+    M: ClientModule,
+{
+    type Target = M;
+
+    fn deref(&self) -> &Self::Target {
+        self.module
+    }
 }
```
Comment (elsirion) [inline fedimint-client/src/lib.rs:587]: Feels a bit dirty but I like it xD

File: fedimint-client/src/lib.rs [hunk 2]
```diff
@@ -564,13 +564,26 @@ const SUPPORTED_CORE_API_VERSIONS: &[fedimint_core::module::ApiVersion] =
...
 
 /// Resources particular to a module instance
-pub struct ClientModuleInstance {
+pub struct ClientModuleInstance<'m, M: ClientModule> {
     /// Instance id of the module
     pub id: ModuleInstanceId,
     /// Module-specific DB
     pub db: Database,
     /// Module-specific API
     pub api: DynModuleApi,
```
Comment (elsirion) [inline fedimint-client/src/lib.rs:573]: Separate issue: Should these be accessible via functions so we can change the way they are stored?

**Review by elsirion — APPROVED** [review]
Great PR! Once this is done for all modules `fedimint-cli` should probably also be changed to only allow modularized operations.

**Comment by elsirion** [comment]
I think you actually broke some tests.

File: fedimint-client/src/lib.rs (hunk 1 above)
Comment (dpc) [reply fedimint-client/src/lib.rs:587]: We could just make it a method or a field, and it wouldn't be that much of a downgrade.

... but if we squint our eyes, we can consider it a smart pointer over module, so I guess it's OK?

File: fedimint-client/src/lib.rs (hunk 2 above)
Comment (dpc) [reply fedimint-client/src/lib.rs:573]: Was thinking the same. Probably yes, so the owner can't assign at least?

Comment (elsirion) [reply fedimint-client/src/lib.rs:573]: I could also imagine just having a normal dbtx reference in there and creating the module dbtx on demand etc.

**Review by justinmoon — APPROVED** [review]

**Pull request merged**